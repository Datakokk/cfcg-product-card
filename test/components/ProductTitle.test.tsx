import React from "react";
import renderer from "react-test-renderer"
import { ProductCard, ProductTitle } from '../../src/components';
import { product1 } from "../data/products";


describe('Test on <ProductTitle /> ', () => { 
    
    test('show display component correctly with custom title', () => { 
        

        const wrapper = renderer.create(
            <ProductTitle title="Custom test Product" className="custom-class"/>
        );

        expect( wrapper.toJSON() ).toMatchSnapshot();
     });

     test('should display component with the product name', () => { 

        const wrapper = renderer.create(
            <ProductCard product={ product1 }>
                {
                    () => (
                        <ProductTitle />
                    )
                }
            </ProductCard>
        );
        expect( wrapper.toJSON() ).toMatchSnapshot();
      })
 })