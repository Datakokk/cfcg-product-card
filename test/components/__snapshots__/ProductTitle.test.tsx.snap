// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <ProductTitle />  should display component with the product name 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - Without image
  </span>
</div>
`;

exports[`Test on <ProductTitle />  show display component correctly with custom title 1`] = `
<span
  className="productDescription custom-class"
>
  Custom test Product
</span>
`;
